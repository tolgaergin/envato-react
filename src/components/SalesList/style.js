import styled from 'styled-components';

export const Item = styled.li `
  position: relative;
  height: 42px;
  padding: 22px 100px 15px 20px;
  border-bottom: 1px solid rgba(205, 206, 210, 0.5);

  &:last-of-type {
    margin-bottom: 20px;

    div::after {
      display: none;
    }
  }

  &:nth-child(even) {
    background-color: #fff;
  }
`;

export const ItemName = styled.span `
  display: block;
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
`;

export const ItemDetail = styled.span `
  display: block;
  font-size: 14px;
  color: #8c8c8c;
  margin-top: 2px;
`;

export const ItemPrice = styled.span `
  position: absolute;
  top: 50%;
  transform: translateY(-50%);
  right: 20px;
  font-size: 22px;
  text-align: right;

  sup {
    position: relative;
    top: -0.4em;
    vertical-align: baseline;
    font-size: 14px;
  }
`;

export const ItemDot = styled.div `
  position: absolute;
  top: 32px;
  right: 86px;
  z-index: 10;

  &::before {
    position: absolute;
    content: '';
    width: 13px;
    height: 13px;
    border-radius: 50%;
    background-color: #cdced2;
    box-shadow:
      inset 0px 0px 0px 3px #fff,
      0px 0px 0px 2px #cdced2;
  }

  &::after {
    position: absolute;
    content: '';
    top: 15px;
    left: 6px;
    width: 1px;
    height: 64px;
    background-color: #cdced2;
  }
`;
