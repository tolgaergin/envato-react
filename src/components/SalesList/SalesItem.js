import React from 'react';
import moment from 'moment';

import { Item, ItemName, ItemDetail, ItemPrice, ItemDot } from './style';

const SalesItem = ({ item, index }) => (
  <Item>
    <ItemName>{item.detail}</ItemName>
    <ItemDetail>
      {item.other_party_country} &#8226; {moment(new Date(item.date)).fromNow()}
    </ItemDetail>
    <ItemPrice><sup>$</sup>{item.price}</ItemPrice>
    <ItemDot />
  </Item>
);

SalesItem.propTypes = {
  item: React.PropTypes.object,
  index: React.PropTypes.string,
};

export default SalesItem;
