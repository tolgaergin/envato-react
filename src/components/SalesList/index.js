import React from 'react';
import Loading from '../Loading';
import SalesItem from './SalesItem';

const SalesList = ({ sales, isFetching }) => (
  isFetching ? <Loading /> : (
    <div className="child">
      <h2>Sales</h2>
      <ul>
        {
          Object.keys(sales).map(key =>
            <SalesItem
              key={key}
              index={key}
              item={sales[key]}
            />
          )
        }
      </ul>
    </div>
  )
);

SalesList.propTypes = {
  sales: React.PropTypes.array,
  isFetching: React.PropTypes.bool,
};

export default SalesList;
