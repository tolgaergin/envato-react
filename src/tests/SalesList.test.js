import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import SalesList from '../components/SalesList/index';

describe('SalesList', () => {
  it('renders and matches snapshot', () => {
    const isFetching = false;
    const sales = [
      {
        detail: 'Pleasure Template',
        price: 20,
        date: '2017-02-16 00:34:36 +1100',
      },
    ];

    const component = renderer.create(
      <SalesList isFetching={isFetching} sales={sales} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
