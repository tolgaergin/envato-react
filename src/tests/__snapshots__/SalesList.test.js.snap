exports[`SalesList renders and matches snapshot 1`] = `
<div
  className="child">
  <ul>
    <li
      className="gSsLRZ">
      <span
        className="gdEpjv">
        Pleasure Template
      </span>
      <span
        className="jZQKrC">
         â€¢ 
        8 days ago
      </span>
      <span
        className="kqPhYY">
        <sup>
          $
        </sup>
        20
      </span>
      <div
        className="krlaZP" />
    </li>
  </ul>
</div>
`;
